File: CPP/selectionSort.cpp
#include <bits/stdc++.h>
using namespace std;

void selectionSort(int arr[], int n)
{
    for (int i = 0; i < n - 1; i++)
    {
        int min_ind = i;
        for (int j = i; j < n; j++)
            if (arr[j] < arr[min_ind])
                min_ind = j;
        swap(arr[i], arr[min_ind]);
    }
}
void printArray(int arr[], int n)
{
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << endl;
}
int main()
{
    int arr[] = {5, 4, 2, 3, 1};
    selectionSort(arr, 5);
    printArray(arr, 5);
}

File: DSA Using JavaScripts/SelectionSort.js
var arr =[8,3,1,2,5,4,7,6]
let n=arr.length

function selectionSort(arr,n){
   for(let i=0;i<n-1;i++){
       let min=i
       for(let j=i+1;j<n;j++){
             if(arr[j]<arr[min]) {
                min =j;
             }
            }
            let temp=arr[min] 
           arr[min]=arr[i]
           arr[i]=temp  
        // swap(arr,min,i)
   }
   return arr
}
console.log(selectionSort(arr,n));